.github/workflows/soak.yml

```diff
@@ -0,0 +1,244 @@
+          - syslog_humio_logs
+          - syslog_log2metric_humio_metrics
+          - syslog_loki
+          - syslog_regex_logs2metric_ddmetrics
+          - syslog_splunk_hec_logs
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+
+      - name: Download baseline-soak artifact
+        uses: actions/download-artifact@v1
+        with:
+          name: baseline-soak
+
+      - name: Run comparison experiment
+        run: |
+          mkdir -p /tmp/${{ matrix.target }}/
+          ./soaks/bin/soak_one.sh "false" ${{ matrix.target }} "comparison" ${{ github.sha }} /tmp/${{ matrix.target }}
+
+      - name: Upload timing captures
+        uses: actions/upload-artifact@v1
+        with:
+          name: ${{ matrix.target}}-captures
+          path: /tmp/${{ matrix.target }}
+
+  analyze-results:
+    name: Soak analysis (${{ matrix.target }})
+    if: ${{ github.actor != 'dependabot[bot]' }}
+    runs-on: [self-hosted, linux, x64, soak] # could be general if we move away from miller
+    needs: [soak-baseline, soak-comparison]
+    strategy:
+      matrix:
+        target:
+          - datadog_agent_remap_datadog_logs
+          - syslog_humio_logs
+          - syslog_log2metric_humio_metrics
+          - syslog_loki
+          - syslog_regex_logs2metric_ddmetrics
+          - syslog_splunk_hec_logs
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+
+      - name: Download captures artifact
+        uses: actions/download-artifact@v1
+        with:
+          name: ${{ matrix.target }}-captures
+
+      - name: Analyze captures
+        run: |
+          ./soaks/bin/analyze_experiment.sh ${{ matrix.target }}-captures
+
+  observer:
+    name: Build and push 'observer' to Github CR
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        with:
+          images: ghcr.io/${{ github.repository }}/soak-observer
+          tags: type=sha, format=long
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
+        with:
+          context: .
+          file: lib/soak/Dockerfile
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          cache-from: type=gha, scope=${{ github.workflow }}
+          cache-to: type=gha, scope=${{ github.workflow }}
```

Comment:
Should we add the same "notify on failure" job for master that we have for other workflows?

Reply:
Ah yeah, good idea. Will do.

Reply:
Hmm, on second thought, up to you. The existing failure notification drops to discord which I, personally, don't interact with much so I wouldn't see that notification. Near term -- #9622 -- we'll have PR feedback on soaks so I'd expect we can bundle failure notification up there.